flink-runtime/src/main/java/org/apache/flink/runtime/operators/hash/CompactingHashTable.java

```diff
@@ -406,9 +403,7 @@ public void insertOrReplaceRecord(T record) throws IOException {
 						return;
 					}
 				}
-				else {
-					numInSegment++;
-				}
+				numInSegment++;
```

Comment:
this is not equivalent to the old implementation, since numInSegent is no longer incremented if "this.buildSideComparator.equalToReference(valueAtPosition)" is true, since we return.